src/main/java/org/elasticsearch/cluster/metadata/RepositoriesMetaData.java

```diff
@@ -80,6 +82,32 @@ public RepositoryMetaData repository(String name) {
         return null;
     }
 
+    @Override
+    public MetaData.Custom readFrom(StreamInput in) throws IOException {
+        return FACTORY.readFrom(in);
+    }
+
+    @Override
+    public void writeTo(StreamOutput out) throws IOException {
+        FACTORY.writeTo(this, out);
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+
+        RepositoriesMetaData that = (RepositoriesMetaData) o;
+
+        return repositories.equals(that.repositories);
```

Comment:
We cannot delegate equals because we need to get underlying repositories list, but in order to do that we need to convert `o` to `RepositoriesMetaData`, which requires checking its type first, and make sure that it's not null, which this code is essentially doing.